### DISCUSSION TITLE
Kafka를 Storage처럼 데이터를 영구 저장하게 사용할 수 있는가?

### DISCUSSION THREAD

**Discussion by sejongjeong** [kyungjunleeme/kafka_study#8] [Q&A]
Kafka를 Storage처럼 데이터를 영구 저장하게 사용할 수 있는가?

**Comment by kyungjunleeme** [comment]
https://developnote-blog.tistory.com/197
관련 설정 내용

**Comment by hellices** [comment]
kafka는 3.6.0버전(이 때는 preview)부터 tiered storage라는 컨셉을 제공하고 있습니다.
이를 통해 장기 보관이 필요한 메시지는 s3로 이관하는 전략을 사용할 수 있습니다.
ref) https://kafka.apache.org/documentation/#tiered_storage

**Reply by kyungjunleeme** [reply]
해당질문에 대한 답변 내용은 아니지만, 인지필요

**Comment by kyungjunleeme** [comment]
log.retention.bytes  # 기본값은 -1 지정시 영구저장
log.retention.hours  # -1로 설정하면 log는 지워지지 않고 broker log directory에 영원히 저장됨

보통 log 정책을 기간 단위로 세우기 때문에 해당 옵션은 우선순위로 고려되지 않습니다

**Reply by kyungjunleeme** [reply]
<추가확인필요>
- 정리하면 broker config 설정이 -> topic 설정에 어떤 영향을 주고 상호간의 관계도 파악할 필요가 있음
- 설정관에 우선순위가 있음.


- https://stackoverflow.com/questions/65507232/kafka-log-segment-bytes-vs-log-retention-hours
해당 글에서는 partition 용량도 한계가 있으니 시간과함께 무한대로 설정해두면, 우선 둘중하나 만족하는 것으로 인해 데이터가 삭제될것이라는 이야기 

broker config <> topic config 간의 관계도 파악
![image](https://github.com/user-attachments/assets/525acc56-918b-4fb0-bdb0-d0bd9998764a)
